File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Gui/App/Mappers/Tenants/TenantAOOModelMapper.ts
import TenantAOOModel = require("../../Models/Tenants/TenantAOOModel");
import BaseMapper = require("../BaseMapper");

class TenantAOOModelMapper extends BaseMapper<TenantAOOModel>{
    constructor() {
        super();
    }
    public Map(source: any): TenantAOOModel {
        let toMap: TenantAOOModel = <TenantAOOModel>{};

        if (!source) {
            return null;
        }
        toMap.UniqueId = source.UniqueId;
        toMap.Name = source.Name;
        toMap.Note = source.Note;
        toMap.CategorySuffix = source.CategorySuffix;
        toMap.Tenants = source.Tenants;
        toMap.TenantTypology = source.TenantTypology;
        return toMap;
    }
}

export = TenantAOOModelMapper;

File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Mapper.Model/Tenants/Implementation/TenantAOOModelMapper.cs
using VecompSoftware.DocSuiteWeb.Entity.Tenants;
using VecompSoftware.DocSuiteWeb.Model.Entities.Tenants;

namespace VecompSoftware.DocSuiteWeb.Mapper.Model.Tenants
{
    public class TenantAOOModelMapper : BaseModelMapper<TenantAOO, TenantAOOModel>, ITenantAOOModelMapper
    {
        public TenantAOOModelMapper()
        {
            
        }

        #region [ Methods ]
        public override TenantAOOModel Map(TenantAOO entity, TenantAOOModel modelTransformed)
        {
            modelTransformed.IdTenantAOO = entity.UniqueId;
            modelTransformed.Name = entity.Name;
            modelTransformed.Note = entity.Note;

            return modelTransformed;
        }
        #endregion
    }
}
